Data/Browncoats.rte/Devices/Shields/Eruptor/Eruptor.lua

```diff
@@ -0,0 +1,164 @@
+		end
+	end
+	
+end
+
+function OnAttach(self, newParent)
+
+	self.equippedInMainHand = false;
+	self:SetOneHanded(true);
+	self.Bashing = false;
+	self.bashActive = false;
+	self.HitsMOs = false	local rootParent = self:GetRootParent();
+	if IsAHuman(rootParent) then
+		if ToAHuman(rootParent).EquippedItem.UniqueID == self.UniqueID then
+			ToAHuman(rootParent):UnequipBGArm();
+			self.equippedInMainHand = true;
+			self:SetOneHanded(false);
+		end
+	end
+	
+end
+
+function ThreadedUpdate(self)
+
+	if self.equippedInMainHand then
+		if not self.bashCoolDown then
+			if self:IsActivated() then
+				if not self.Bashing then
+					self.bashWindupSound:Play(self.Pos);
+					self.bashSwingSound:Play(self.Pos);
+					self.hitMOTable = {};
+					self.Bashing = true;
+					self.bashTimer:Reset();
+					self.StanceOffset = self.bashWindupStanceOffset;
+					self.SharpStanceOffset = self.bashWindupStanceOffset;
+				end
+			end
+		elseif self.bashTimer:IsPastSimMS(self.bashCoolDownTime) then
+			self.bashCoolDown = false;
+		end	
+		
+		if self.Bashing then
+			if self.bashTimer:IsPastSimMS(self.bashWindupTime + self.bashActiveTime) then
+				self.Bashing = false;
+				self.bashActive = false;
+				self.bashTimer:Reset();
+				self.bashCoolDown = true;
+				self.StanceOffset = self.origStanceOffset;
+				self.SharpStanceOffset = self.origSharpStanceOffset;
+			elseif self.bashTimer:IsPastSimMS(self.bashWindupTime) then
+				local topRayPos = self.Pos + Vector(0, 12):RadRotate(self.RotAngle);
+				local midRayPos = self.Pos;
+				local bottomRayPos = self.Pos + Vector(0, -12):RadRotate(self.RotAngle);
+				
+				local trace = Vector(8 * self.FlipFactor, 0):RadRotate(self.RotAngle);
+				
+				--PrimitiveMan:DrawLinePrimitive(0, topRayPos, topRayPos + trace, 255);
+				--PrimitiveMan:DrawLinePrimitive(0, midRayPos, midRayPos + trace, 255);
+				--PrimitiveMan:DrawLinePrimitive(0, bottomRayPos, bottomRayPos + trace, 255);
+				
+				local rayID = SceneMan:CastMORay(topRayPos, trace, self.ID, self.Team, rte.grassID, false, 2);
+				local rayID = rayID == rte.NoMOID and SceneMan:CastMORay(midRayPos, trace, self.ID, self.Team, rte.grassID, false, 2) or rayID;
+				local rayID = rayID == rte.NoMOID and SceneMan:CastMORay(bottomRayPos, trace, self.ID, self.Team, rte.grassID, false, 2) or rayID;
+				if rayID ~= rte.NoMOID then
+					local MO = MovableMan:GetMOFromID(rayID);
+					local rootMO = MO:GetRootParent();
+					if not self.hitMOTable[rootMO.UniqueID] and IsActor(rootMO) then
+						self.hitMOTable[rootMO.UniqueID] = true;
+						self.hitActor = ToActor(rootMO);
+						self.bashImpactSound:Play(self.Pos);
+						self:RequestSyncedUpdate();
+					end
+				end
+
+				if not self.bashActive then
+					self.bashActive = true;
+				end
+				self.StanceOffset = self.bashActiveStanceOffset;
+				self.SharpStanceOffset = self.bashActiveStanceOffset;
+			end
+		end
+	end
+
+end
+
+function SyncedUpdate(self)
+
+	if MovableMan:ValidMO(self.hitActor) then
```

Comment:
self.hitActor isn't being reset back to nil anywhere? This means another script that causes a synced update will cause this to happen again.

(note - I want to change this in future so that SyncedUpdate is sandboxed to each script, but it's still just a good idea for code cleanliness)